src/test/java/io/supertokens/test/oauth2/StorageTests.java

```diff
@@ -63,27 +67,235 @@ public void testCreateOAuth2ClientTransaction() throws Exception {
+                        storage.createOAuth2Client_Transaction(new AppIdentifier(null, null), con,
+                                getOAuth2Client("exp2", "exp2",oAuth2Client.clientSecretHash));
+                    } catch (DuplicateOAuth2ClientSecretHash | DuplicateOAuth2ClientIdException |
+                             TenantOrAppNotFoundException e) {
+                        throw new StorageTransactionLogicException(e);
+                    }
+                    return null;
+                });
+            } catch (StorageTransactionLogicException e) {
+                error = e.actualException;
+            }
+
+            Assert.assertNotNull(error);
+            assert (error instanceof DuplicateOAuth2ClientSecretHash);
+        }
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testGetOAuth2ClientByIdExceptions() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        OAuth2SQLStorage storage = (OAuth2SQLStorage) StorageLayer.getStorage(process.getProcess());
+
+        Exception error = null;
+        try {
+            storage.getOAuth2ClientById(new AppIdentifier(null, null), "invalid-client-id" );
+        } catch (Exception e) {
+            error = e;
+        }
+
+        Assert.assertNotNull(error);
+        assert (error instanceof UnknownOAuth2ClientIdException);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+    @Test
+    public void testCreateOAuth2Scope() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        OAuth2SQLStorage storage = (OAuth2SQLStorage) StorageLayer.getStorage(process.getProcess());
+        List<String> scopes = new ArrayList<>();
+        scopes.add("profile");
+        scopes.add("home_page");
+
+        scopes.forEach(
+                it -> {
+                    try {
+                        storage.createOAuth2Scope(new AppIdentifier(null, null), it);
+                    } catch (DuplicateOAuth2ScopeException | TenantOrAppNotFoundException | StorageQueryException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+        );
+
+        List<String> getOAuth2ScopesFromAppId = storage.getOAuth2Scopes(new AppIdentifier(null, null));
```

Comment:
since this function is implemented already in the plugin, add tests for this too:
- getting scopes when non exist should return an empty list
- getting scopes when some exist should return the right list

Reply:
changes made